zebra-state/src/lib.rs

```diff
@@ -27,20 +27,58 @@ pub mod on_disk;
 #[serde(deny_unknown_fields)]
 pub struct Config {
     /// The root directory for the state storage
-    pub path: PathBuf,
+    pub path: Option<PathBuf>,
 }
 
 impl Config {
+    /// Generate the appropriate `sled::Config` based on the provided
+    /// `zebra_state::Config`.
+    ///
+    /// # Details
+    ///
+    /// This function should panic if the user of `zebra-state` doesn't configure
+    /// a directory to store the state.
+    ///
+    /// Currently this is only partially implemented by leveraging a `SpanTrace`
+    /// to print a TODO to the user which isn't as nice as what we'd eventually
+    /// want but it definitely gets the point across.
+    ///
+    /// # Current Format
+    ///
+    /// <pre><font color="#CC0000">The application panicked (crashed).</font>
+    /// Message:  <font color="#06989A">called `Option::unwrap()` on a `None` value</font>
+    /// Location: <font color="#75507B">zebra-state/src/lib.rs</font>:<font color="#75507B">40</font>
+    ///
+    /// Backtrace omitted.
+    ///
+    /// Run with RUST_BACKTRACE=1 environment variable to display it.
+    /// Run with RUST_BACKTRACE=full to include source snippets.
+    ///   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+    ///
+    ///    0: <font color="#F15D22">zebra_state::sled_config::comment</font> with <font color="#34E2E2">text=TODO(jlusby): Replace this unwrap with a nice user-facing</font>
+    /// <font color="#34E2E2">   explaining error that the cache dir must be specified</font>
+    ///       at <font color="#75507B">zebra-state/src/lib.rs</font>:<font color="#75507B">36</font></pre>
```

Comment:
This looks gross in the source but when you run `cargo doc --include-private-items` it looks SO NICE

`copy as HTML` is the best feature of my terminal emulator.

Reply:
Look at thissss

![Screenshot from 2020-07-22 11-49-46](https://user-images.githubusercontent.com/1993852/88216079-7785f700-cc11-11ea-96c1-8e06db3a77c2.png)

Reply:
Ignore the out of order typo, I already fixed it

Reply:
I'm not sure this is worth including or if I should just rip out the spandoc stuff here and use an `expect` call, but I wanted to demonstrate the capability because I think its swell.

Reply:
I think expect might be easier to read?